csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -868,6 +941,12 @@ public Settings(bool formatDefaultValues, TypeRegistry typeRegistry) : this(form
             /// </summary>
```

Comment:
All the "With" methods need to be updated to preserve the existing indentation. (`settings.WithFormatDefaultValues(true).WithIndentation("  ")` should be equivalent to `settings.WithIndentation("  ").WithFormatDefaultValues(true)`)

Reply:
Good catch. Covered this with new unit tests and implemented the fix.